src/Plugin/Sampler/Bundle.php

```diff
@@ -174,4 +186,32 @@ protected function isBaseField(FieldDefinitionInterface $fieldConfig) {
     return ($fieldConfig instanceof BaseFieldDefinition || $fieldConfig instanceof BaseFieldOverride);
   }
 
+  /**
+   * Get all supported field definitions.
+   *
+   * Filters all fields that have a not supported type. Supported types are
+   * defined in sampler settings.
+   *
+   * @param string $entityTypeId
+   *   The entity type.
+   * @param string $bundle
+   *   The bundle.
+   *
+   * @return array
+   *   The field definitions.
+   */
+  protected function getSupportedFieldDefinitions(string $entityTypeId, string $bundle) {
+    $fieldDefinitions = $this->entityFieldManager->getFieldDefinitions($entityTypeId, $bundle);
+    $supportedFieldTypes = array_flip($this->samplerSettings->get('supported_field_types'));
+
+    $supportedFields = [];
+    foreach ($fieldDefinitions as $fieldName => $fieldConfig) {
```

Comment:
It's not so much relevant, but if you are already working on this branch, then you could maybe apply `array_filter` on `fieldDefinitions`, it would be cleaner.

Other then that, everything looks good.